### PR TITLE
feat: add support for route_optimized parameter in databricks deployment client

### PR DESCRIPTION
**Pull request by prabhatkgupta**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/prabhatkgupta/mlflow/pull/13222?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13222/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13222
```

</p>
</details>

### Related Issues/PRs

 #13221

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [x] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?
No

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [x] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

**Comment by B-Step62** [comment]
Thank you for submitting the PR, @prabhatkgupta! Could you fill the PR template?

File: mlflow/deployments/databricks/__init__.py [hunk 1]
```diff
@@ -336,6 +336,7 @@ def create_endpoint(self, name, config=None):
         Args:
             name: The name of the serving endpoint to create.
             config: A dictionary containing the configuration of the serving endpoint to create.
+            kwargs: Extra arguments for serving endpoint
```
[the remark this answers is not in the document]
Comment (prabhatkgupta) [reply mlflow/deployments/databricks/__init__.py:339]: Sure, that would be much better way

**Comment by prabhatkgupta** [comment]
> Thank you for submitting the PR, @prabhatkgupta! Could you fill the PR template?

Since this is my first contribution, I would ask about what fields are required in PR template ?? @B-Step62

File: mlflow/deployments/databricks/__init__.py [hunk 2]
```diff
@@ -377,12 +379,13 @@ def create_endpoint(self, name, config=None):
...
 
         """
+        route_optimized = kwargs.get("route_optimized", False)
         config = config.copy() if config else {}  # avoid mutating config
         extras = {}
         for key in ("tags", "rate_limits"):
             if tags := config.pop(key, None):
                 extras[key] = tags
-        payload = {"name": name, "config": config, **extras}
+        payload = {"name": name, "config": config, "route_optimized": route_optimized, **extras}
```
[the remark this answers is not in the document]
Comment (prabhatkgupta) [reply mlflow/deployments/databricks/__init__.py:387]: This is an independent parameter which lies outside the scope of config, that's why passed as a separate named parameter @B-Step62

**Comment by B-Step62** [comment]
> Since this is my first contribution, I would ask about what fields are required in PR template ??

Sure! Basically all check box questions need to be filled. Please let me know if you are unclear about any question, but you don't have to be 100% accurate🙂 (it's more for labeling and analysis purpose).

**Comment by prabhatkgupta** [comment]
> > Since this is my first contribution, I would ask about what fields are required in PR template ??
> 
> Sure! Basically all check box questions need to be filled. Please let me know if you are unclear about any question, but you don't have to be 100% accurate🙂 (it's more for labeling and analysis purpose).

@B-Step62 I have filled the PR description, please check now

File: mlflow/deployments/databricks/__init__.py [hunk 3]
```diff
@@ -336,6 +336,7 @@ def create_endpoint(self, name, config=None):
         Args:
             name: The name of the serving endpoint to create.
             config: A dictionary containing the configuration of the serving endpoint to create.
+            route_optimized: A boolean which defines whether databricks serving endpoint in optimized for routing traffic
```
Comment (B-Step62) [inline mlflow/deployments/databricks/__init__.py:339]:
```suggestion
            route_optimized: A boolean which defines whether databricks serving endpoint
                in optimized for routing traffic. Refer to the following documentation for more details.
                https://docs.databricks.com/en/machine-learning/model-serving/route-optimization.html
```

**Comment by prabhatkgupta** [comment]
@B-Step62 Pushed commit to update documentation link to exact code snippet 
> https://docs.databricks.com/en/machine-learning/model-serving/route-optimization.html#enable-route-optimization-on-a-feature-serving-endpoint

**Review by B-Step62 — APPROVED** [review]
The change looks good to me, thanks @prabhatkgupta for the enhancement!!

**Pull request merged**